src/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,240 @@
+    const currentProductPrice = on_discount ? getDiscountedPrice(price, discount_percentage) : price;
+    return previousProductPrice > currentProductPrice ? currentProductPrice : previousProductPrice;
+  }, products[0].price)
+}
+
+function getMaximumProductPrice(products) {
+
+  if(products && products.length === 0) 
+  return 0;
+
+  return products.reduce((previousProductPrice, {on_discount, discount_percentage, price}) => {
+    const currentProductPrice = on_discount ? getDiscountedPrice(price, discount_percentage) : price;
+    return previousProductPrice < currentProductPrice ? currentProductPrice : previousProductPrice;
+  }, products[0].price)
+}
+
+const featureValueToFeatureName = feature => feature.split('_').map(word => word[0].toUpperCase() + word.substr(1)).join(' ')
+
+const sortingCriteriaValueToSortingCriteriaName = (sortingCriteria) => {
+  const sortingCriteriaName = sortingCriteria.split('-').map(word => word[0].toUpperCase() + word.substr(1));
+  sortingCriteriaName[0] = sortingCriteriaName[0] + ":";
+  return sortingCriteriaName.join(' ');
+}
+
+export const Sidebar = () => {
+
+    const [categories, setCategories] = useState();
+    const [features, setFeatures] = useState();
+    const {state, dispatch} = useProductsAndFiltersData()
+
+
+    useEffect(() => {
+      console.log(state);
+      (async() => {
+        const recievedCategories = await axios.get(`${BACKEND_URL}/categories`);
+        setCategories(recievedCategories.data);
+        
+        const recievedFeatures = await axios.get(`${BACKEND_URL}/features`);
+        setFeatures(recievedFeatures.data);
+      })();
+    },[])
+
+    const maximumProductPrice = Math.ceil(getMaximumProductPrice(state.allProducts));
+    const minimumProductPrice = Math.floor(getMinimumProductPrice(state.allProducts));
+    const averageProductPrice = Math.floor(minimumProductPrice + (maximumProductPrice - minimumProductPrice) / 2.0);
+    const priceSliderRef = useRef();
+
+    console.log(minimumProductPrice, maximumProductPrice)
+
+    
+    return (
+        <aside className="sidebar-container fixed-position">
+        <div className="filters-heading-container row-container">
+          <h3>Clear Filters</h3>
+        </div>
+
+        <h2 className="sidebar-heading">Price</h2>
+        <div className="price-slider">
+          <div className="price-range-values-container width-100-percent">
+            <h3>{minimumProductPrice}</h3>
+            <h3>{averageProductPrice}</h3>
+            <h3>{maximumProductPrice}</h3>
+          </div>
+          <input
+            type="range"
+            min={minimumProductPrice}
+            max={maximumProductPrice}
+            defaultValue={maximumProductPrice}
+            id="price-slider"
+            className="slider width-100-percent"
+            ref={priceSliderRef}
+            onChange={
+              (event) => {
+                dispatch({
+                  ...state,
```

Comment:
Not an expert, but I don't feel it's necessary to use the spread operator and pass all the state attributes when calling the dispatch. The convention that I have come across is a type and payload passed to the dispatch only.